ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -1844,14 +1853,18 @@ impl<'a> Assemble<'a> {
                         reg_alloc::Register::GP(r) => {
                             let [_] = self.ra.assign_gp_regs(
                                 &mut self.asm,
-                                iidx,
+                                // This tells the register allocator to assign this register
+                                // to the instruction at `iidx` so we can look it up later. Since
```

Comment:
The "since we are at the end" bit is dangerous: the register allocator does (must!) take into account `iidx` when making decisions about what to spill etc. By passing in `0` here, I think weird things will happen. Is there any reason not to pass `iidx` here (and below)?

Reply:
We fixed this in a commit (https://github.com/ykjit/yk/pull/1510/commits/860f69c87b4009b32a680b9cfeb8ecf7a4341f1b).